processor/blockprocessor/handler_processor.go

```diff
@@ -0,0 +1,95 @@
+package blockprocessor
+
+import (
+	"fmt"
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/rpcs"
+	indexerledger "github.com/algorand/indexer/processor/eval"
+	"github.com/sirupsen/logrus"
+)
+
+type blockProcessor struct {
+	handler func(block *ledgercore.ValidatedBlock) error
+	ledger  *ledger.Ledger
+	logger  *logrus.Logger
+}
+
+func (proc *blockProcessor) AddGenesisBlock() error {
+	if proc.handler != nil && uint64(proc.ledger.Latest()) == 0 {
+		blk, err := proc.ledger.Block(0)
+		if err != nil {
+			return fmt.Errorf("addGenesisBlock() err: %w", err)
+		}
+		vb := ledgercore.MakeValidatedBlock(blk, ledgercore.StateDelta{})
+		err = proc.handler(&vb)
+		if err != nil {
+			return fmt.Errorf("addGenesisBlock() handler err: %w", err)
+		}
+	}
+	return nil
+}
+
+// Process a raw algod block
+func (proc *blockProcessor) Process(blockCert *rpcs.EncodedBlockCert) error {
```

Comment:
I may be missing something. why do I seem to see similar Process logic repeated in multiple places?